### ISSUE TITLE
Cannot register an Effect when using the templated PlatformEffect

### ISSUE THREAD

**Issue by inimirpaz** [dotnet/maui#16517]
### Description

Hello everyone,

I was trying to upgrade an effect I had in a Xamarin project to .NET MAUI (see this for reference https://github.com/xamarin/XamarinCommunityToolkit/blob/65b048ede7a3bb9f16651d9f8fbdd95e761d8aa2/src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/PlatformTouchEffect.android.cs )
In the native Android implementation I made the class inherit from `PlatformEffect<TContainer, TControl>` in the form of `PlatformEffect<Android.Views.ViewGroup, Android.Views.View>`

The problem arises when porting it to .NET MAUI because I am offered a choice with no apparent solution in either case.

1)
I can make it inherit from `Microsoft.Maui.Controls.Platform.PlatformEffect<Android.Views.ViewGroup, Android.Views.View>` as expected, BUT when I try to register it to the app builder with:
```
.ConfigureEffects(effects => {
	effects.Add<MyRoutingEffect, MyPlatformEffect>();
})
```
I get a
`CS0311: The type 'MyApp.Effects.MyPlatformEffect' cannot be used as type parameter 'TPlatformEffect' in the generic type or method 'IEffectsBuilder.Add<TEffect, TPlatformEffect>()'. There is no implicit reference conversion from 'MyApp.Effects.MyPlatformEffect' to 'Microsoft.Maui.Controls.Platform.PlatformEffect'.`

2)
To make the effect registering work I can make the effect inherit from `Microsoft.Maui.Controls.Platform.PlatformEffect` instead, BUT by doing this I have no longer access to the ViewGroup property as the TContainer obviously ( https://github.com/dotnet/maui/blob/e6a4334ad364a41ac730b8f797fbc137f3f74bdc/src/Controls/src/Core/Platform/PlatformEffect.cs#L17 ), which is a must for the effect to work.

### Version with bug

7.0.92

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

**Comment by jfversluis** [comment]
Could you please verify this is still an issue with the latest version in .NET 8? If it does and you're still hitting this, please provide a [reproduction](https://github.com/dotnet/maui/blob/main/.github/repro.md) project which shows this issue so we can have a look.

**Comment by inimirpaz** [comment]
Back in .NET 7 I slightly changed approach.
I went full on preprocessor directives:
the routing is inheriting from `Microsoft.Maui.Controls.RoutingEffect`, the effect is inherithing from `Microsoft.Maui.Controls.Platform.PlatformEffect` and its body is selected by the specific platform at compile time via directives. To access the ViewGroup in Android I ended up using `ViewGroup? Group => Container as ViewGroup ?? Control as ViewGroup;`.
